leakcanary-android-core/src/main/java/leakcanary/EventListener.kt

```diff
@@ -86,6 +86,14 @@ fun interface EventListener {
         showIntent: Intent
       ) : HeapAnalysisDone<HeapAnalysisFailure>(uniqueId, heapAnalysis, showIntent)
     }
+
+    class NoRetainedObjectFound(
```

Comment:
@pyricau I'm considering moving the notifications to the `NotificationEventListener.kt` and creating custom events. 
I noticed that so far Events are Heap Dump based. Not sure if we want to create new events that are not exactly based on Heap Dump's timeline.

Thoughts ?

Reply:
I've refrained from that because I need to think a bit more deeply about this. The listener approach doesn't scale super well.

Reply:
The changes so far do seem fairly reasonable I think.